airbyte-integrations/connectors/source-posthog/source_posthog/source.py

```diff
@@ -0,0 +1,131 @@
+
+        checkpoint_interval = stream_instance.state_checkpoint_interval
+        slices = stream_instance.stream_slices(
+            cursor_field=configured_stream.cursor_field, sync_mode=SyncMode.incremental, stream_state=stream_state
+        )
+        for slice_ in slices:
+            record_counter = 0
+            records = stream_instance.read_records(
+                sync_mode=SyncMode.incremental,
+                stream_slice=slice_,
+                stream_state=stream_state,
+                cursor_field=configured_stream.cursor_field or None,
+            )
+            for record_data in records:
+                stream_state = stream_instance.get_updated_state(stream_state, record_data)
+                if isinstance(stream_state, tuple):
+                    stream_state = stream_state[1]
+                    break
+                record_counter += 1
+                yield self._as_airbyte_record(stream_name, record_data)
+
+                if checkpoint_interval and record_counter % checkpoint_interval == 0:
+                    yield self._checkpoint_state(stream_name, stream_state, connector_state, logger)
+
+            yield self._checkpoint_state(stream_name, stream_state, connector_state, logger)
+
+    def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
+        try:
+            session = requests.Session()
+            session.headers["Authorization"] = "Bearer %s" % config["api_key"]
+            session.headers["Content-Type"] = "application/json"
+            session.headers["User-Agent"] = "posthog-python/1.4.0"
+            url_base = "https://app.posthog.com/"
+            response = session.get(url_base + "api/user")
+            if response.status_code != 200:
+                resp_json = response.json()
+                err_message = resp_json["detail"] if "detail" in resp_json else "unknown error"
+                raise Exception(err_message)  # return with a python traceback in the log
+
+            _ = dateutil.parser.isoparse(config["start_date"])
+            return True, None
+        except Exception as e:
+            return False, repr(e)
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        authenticator = TokenAuthenticator(token=config["api_key"])
+        streams_ = [
+            Annotations(authenticator=authenticator),
+            Cohorts(authenticator=authenticator),
+            Elements(authenticator=authenticator),
+            Events(start_date=config["start_date"], authenticator=authenticator),
+            EventsSessions(authenticator=authenticator),
+            FeatureFlags(authenticator=authenticator),
+            Insights(authenticator=authenticator),
+            InsightsPath(authenticator=authenticator),
+            InsightsSessions(authenticator=authenticator),
+            Persons(authenticator=authenticator),
+            Trends(authenticator=authenticator),
+        ]
+
+        return streams_
```

Comment:
Why do we need the `streams_ `variable, we can immediately return the list of streams?